protostar-cli/src/starknet_commands/invoke.rs

```diff
@@ -0,0 +1,64 @@
+use anyhow::{Context, Result};
+use clap::Args;
+
+use starknet::accounts::{Account, Call, SingleOwnerAccount};
+use starknet::core::types::FieldElement;
+use starknet::core::utils::get_selector_from_name;
+use starknet::providers::jsonrpc::HttpTransport;
+use starknet::providers::JsonRpcClient;
+use starknet::signers::LocalWallet;
+
+#[derive(Args)]
+#[command(about = "Invoke a contract on starknet")]
+pub struct Invoke {
+    /// Address of contract to invoke
+    #[clap(long)]
+    pub contract_address: String,
+
+    /// Name of the function to invoke
+    #[clap(long)]
+    pub entry_point_name: String,
+
+    /// Calldata for the invoked function
+    #[clap(long)]
+    pub calldata: Vec<String>,
+
+    /// Max fee for the transaction. If not provided, max fee will be automatically estimated
+    #[clap(long)]
+    pub max_fee: Option<String>,
+}
+
+pub async fn invoke(
```

Comment:
General question: do you guys think we should additionally validate inputs for commands?

Reply:
What do you mean by `validate`? Isn't `.context` enough for now?

Reply:
I think having proper error handling in code is enough, clap should ensure that at least types of the inputs match anyways.